tests/suites/os/tests/safe-reboot/index.js

```diff
@@ -125,21 +125,15 @@ module.exports = {
 					"Safe reboot waiting on application locks"
 				);
 
-				targetState = {
-					local: {
-						name: 'local',
-						config: {
-							SUPERVISOR_PERSISTENT_LOGGING: 'true',
-							SUPERVISOR_LOCAL_MODE: 'true',
-							SUPERVISOR_OVERRIDE_LOCK: 'true',
-						},
-						apps: {},
-					},
-					dependent: {
-						apps: [],
-						devices: [],
-					},
-				};
+				// Get the current state first - we need this as we must append it, rather than just blindly post a new
+				// target state, to avoid overwriting any existing configuration
+				let state = await request({
```

Comment:
We seem to have introduced a new race condition here, where rpi device types often fail this request with ECONNRESET

```
leviathan-client-1  | [2024-09-03T13:43:07.616Z][a20fc47-os]         # Subtest: Override update locks
leviathan-client-1  | [2024-09-03T13:43:07.616Z][a20fc47-os]             ok 1 - Safe reboot waiting on application locks
leviathan-client-1  | [2024-09-03T13:43:07.822Z][a20fc47-os]             not ok 2 - Error: read ECONNRESET
leviathan-client-1  | [2024-09-03T13:43:07.823Z][a20fc47-os]               ---
leviathan-client-1  |               stack: >
leviathan-client-1  | [2024-09-03T13:43:07.823Z][a20fc47-os]                 new RequestError
leviathan-client-1  | [2024-09-03T13:43:07.823Z][a20fc47-os]                 (/data/suite/node_modules/request-promise-core/lib/errors.js:14:15)
leviathan-client-1  | [2024-09-03T13:43:07.823Z][a20fc47-os] 
leviathan-client-1  | [2024-09-03T13:43:07.823Z][a20fc47-os]                 Request.plumbing.callback (/data/suite/node_modules/request-promise-core/lib/plumbing.js:87:29)
leviathan-client-1  | [2024-09-03T13:43:07.823Z][a20fc47-os] 
leviathan-client-1  | [2024-09-03T13:43:07.823Z][a20fc47-os]                 Request.RP$callback [as _callback] (/data/suite/node_modules/request-promise-core/lib/plumbing.js:46:31)
leviathan-client-1  | [2024-09-03T13:43:07.823Z][a20fc47-os] 
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]                 self.callback (/data/suite/node_modules/request/request.js:185:22)
leviathan-client-1  |               
leviathan-client-1  |                 Request.onRequestError (/data/suite/node_modules/request/request.js:877:8)
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]               at:
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]                 line: 14
leviathan-client-1  |                 column: 15
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]                 file: /data/suite/node_modules/request-promise-core/lib/errors.js
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]                 function: RequestError
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]               type: RequestError
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]               cause:
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]                 &a1 !error
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]                 name: Error
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]                 message: read ECONNRESET
leviathan-client-1  | [2024-09-03T13:43:07.824Z][a20fc47-os]                 stack: |-
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]                   Error: read ECONNRESET
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]                       at TCP.onStreamRead (internal/stream_base_commons.js:209:20)
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]                 errno: -104
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]                 code: ECONNRESET
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]                 syscall: read
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]               error: *a1
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]               options:
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]                 method: GET
leviathan-client-1  | [2024-09-03T13:43:07.825Z][a20fc47-os]                 json: true
leviathan-client-1  | [2024-09-03T13:43:07.826Z][a20fc47-os]                 uri: http://127.0.0.1:48484/v2/local/target-state
leviathan-client-1  | [2024-09-03T13:43:07.828Z][a20fc47-os]                 callback: !function |-
leviathan-client-1  | [2024-09-03T13:43:07.828Z][a20fc47-os]                   "RP$callback"
leviathan-client-1  | [2024-09-03T13:43:07.829Z][a20fc47-os]                   function RP$callback(err, response, body) {
leviathan-client-1  | [2024-09-03T13:43:07.829Z][a20fc47-os]                               plumbing.callback.call(self, err, response, body);
leviathan-client-1  |                           }
leviathan-client-1  | [2024-09-03T13:43:07.829Z][a20fc47-os]                 transform: null
leviathan-client-1  | [2024-09-03T13:43:07.830Z][a20fc47-os]                 simple: true
leviathan-client-1  | [2024-09-03T13:43:07.830Z][a20fc47-os]                 resolveWithFullResponse: false
leviathan-client-1  | [2024-09-03T13:43:07.830Z][a20fc47-os]                 transform2xxOnly: false
leviathan-client-1  |               response: null
leviathan-client-1  | [2024-09-03T13:43:07.830Z][a20fc47-os]               test: Override update locks
leviathan-client-1  | [2024-09-03T13:43:07.831Z][a20fc47-os]               source: |2
leviathan-client-1  | [2024-09-03T13:43:07.831Z][a20fc47-os]                     if (Error.captureStackTrace) { // required for non-V8 environments
leviathan-client-1  | [2024-09-03T13:43:07.831Z][a20fc47-os]                         Error.captureStackTrace(this);
leviathan-client-1  |                 --------------^
leviathan-client-1  | [2024-09-03T13:43:07.832Z][a20fc47-os]                     }
leviathan-client-1  |               ...
leviathan-client-1  | [2024-09-03T13:43:07.840Z][a20fc47-os] 
leviathan-client-1  | [2024-09-03T13:43:07.841Z][a20fc47-os]             Bail out! Error: read ECONNRESET
leviathan-client-1  | [2024-09-03T13:43:07.843Z][a20fc47-os] Bail out! Error: read ECONNRESET

```